src/streamlink/webbrowser/chromium.py

```diff
@@ -0,0 +1,178 @@
+import os
+from contextlib import asynccontextmanager
+from pathlib import Path
+from typing import AsyncGenerator, List, Optional, Union
+
+import trio
+
+from streamlink.compat import is_darwin, is_win32
+from streamlink.plugin.api import validate
+from streamlink.session import Streamlink
+from streamlink.utils.socket import find_free_port_ipv4, find_free_port_ipv6
+from streamlink.webbrowser.webbrowser import Webbrowser
+
+
+class ChromiumWebbrowser(Webbrowser):
+    @classmethod
+    def names(cls) -> List[str]:
+        return [
+            "chromium",
+            "chromium-browser",
+            "chrome",
+            "google-chrome",
+            "google-chrome-stable",
+        ]
+
+    @classmethod
+    def fallback_paths(cls) -> List[Union[str, Path]]:
+        if is_win32:
+            return [
+                str(Path(base) / sub / "chrome.exe")
+                for sub in (
+                    "Google\\Chrome\\Application",
+                    "Google\\Chrome Beta\\Application",
+                    "Google\\Chrome Canary\\Application",
+                )
+                for base in [os.getenv(env) for env in (
+                    "PROGRAMFILES",
+                    "PROGRAMFILES(X86)",
+                    "LOCALAPPDATA",
+                )]
+                if base is not None
+            ]
+
+        if is_darwin:
+            return [
+                "/Applications/Chromium.app/Contents/MacOS/Chromium",
+                str(Path.home() / "Applications/Chromium.app/Contents/MacOS/Chromium"),
+                "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
+                str(Path.home() / "Applications/Google Chrome.app/Contents/MacOS/Google Chrome"),
+            ]
+
+        return []
```

Comment:
What do you think about preferring Chromium here as the first choice over Google Chrome? I'd rather we try to default to the less evil option first if we can. Same note about the tests but I'm not going to mention it there.

Reply:
I am not aware of any Chromium builds on Windows. If there are any, what are their default paths?

Reply:
There are some projects like https://chromium.woolyss.com/ which provide various "google free" builds but you're right there's nothing I can find that has a default path. That's what I get for not having used Chrome or Chromium in almost a decade I guess. 🤷‍♂️